### PR TITLE
figured out a working accessibility test for login form errors

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

test that there's an accessible `alert` element on the page when a user gets an error trying to log in.

after follow up conversation and fiddling, also attempts to make the login form more accessible (specifically the error message when login fails).  see also #2376 

## How was this change tested?

this just augments an existing feature test

## Which documentation and/or configurations were updated?

n/a

### CODE REVIEW

**Comment by jmartin-sul** [comment]
tiny follow on to #2382

**Comment by jmartin-sul** [comment]
er, actually, i realized i can make this a bit more concise, will fix and un-draft shortly...

**Comment by jmartin-sul** [comment]
ugh, i got worried that my test wasn't doing what i expected, and i was right, because both of these expectations passed too:
```js
await screen.findByRole('alert', { description: /incorrect username or abcd/i })
await screen.findByRole('alert', { foo: /incorrect username or abcd/i })
```

which leads me to believe that despite `description` being part of the accessibility tree:  https://developer.mozilla.org/en-US/docs/Glossary/AOM

it doesn't seem to be a param for react testing library's `*ByRole` queries:  https://testing-library.com/docs/dom-testing-library/api-queries#byrole
```js
getByRole(
  container: HTMLElement, // if you're using `screen`, then skip this argument
  role: TextMatch,
  options?: {
    exact?: boolean = true,
    hidden?: boolean = false,
    name?: TextMatch,
    normalizer?: NormalizerFn,
    selected?: boolean,
    checked?: boolean,
    pressed?: boolean,
    queryFallbacks?: boolean,
  }): HTMLElement
```

amending back to the less terse but more precise version.

**Comment by jmartin-sul** [comment]
3rd time's a charm?

File: __tests__/feature/userAuthentication.test.js [hunk 1]
```diff
@@ -71,12 +71,17 @@ describe('user authentication', () => {
...
+
     // try to login
     fireEvent.change(screen.getByLabelText('User name'), { target: { value: 'baz.le.quux@example.edu' } })
     fireEvent.change(screen.getByLabelText('Password'), { target: { value: 'password1' } })
     fireEvent.click(screen.getByRole('button', { name: 'Login' }))
 
     // an error message should be presented
     await screen.findByText(/Incorrect username or password./)
+    // ...and an accessible alert should be present now that a failed login attempt was made (TODO: figure out querying on alert text)
+    screen.getByRole('alert')
```
Comment (justinlittman) [inline __tests__/feature/userAuthentication.test.js:85]: I think you want `screen.getByRole('alert', { name: 'ALERT TEXT' })` to query on alert text.

Comment (jmartin-sul) [reply __tests__/feature/userAuthentication.test.js]: so i'd tried a couple variations of that, and the expectation always failed when i tried to check the text in addition to the alert role.

while working on #2276 later yesterday evening, i discovered the accessibility debugger, and i went back and used it to look into this problem.  i noticed there wasn't actually a `name` property:
Screen Shot 2020-08-28 at 2 35 35 PM

later, when debugging my WIP on #2276, i noticed the use of `aria-label` in the search form, and thought i'd try applying it to the login form's error message div.  and that seemed to work:
Screen Shot 2020-08-28 at 2 36 28 PM

i just pushed up a commit with the failing test that i was writing/you're suggesting, about to push up a commit with the fix to the login form (hopefully).

File: __tests__/feature/userAuthentication.test.js [hunk 2]
```diff
@@ -71,12 +71,15 @@ describe('user authentication', () => {
...
+
     // try to login
     fireEvent.change(screen.getByLabelText('User name'), { target: { value: 'baz.le.quux@example.edu' } })
     fireEvent.change(screen.getByLabelText('Password'), { target: { value: 'password1' } })
     fireEvent.click(screen.getByRole('button', { name: 'Login' }))
 
-    // an error message should be presented
-    await screen.findByText(/Incorrect username or password./)
+    // ...and an accessible alert should be present now that a failed login attempt was made
+    await screen.findByRole('alert', { name: 'Incorrect username or password.' })
```
Comment (jmartin-sul) [inline __tests__/feature/userAuthentication.test.js:83]: fails prior to the second commit in this PR:
Screen Shot 2020-08-28 at 2 42 33 PM

**Comment by jmartin-sul** [comment]
grr, line length...

**Comment by jmartin-sul** [comment]
3rd commit on the fourth approach is a charm, as they say.

**Comment by jmartin-sul** [comment]
couldn't resist poking at this further since i thought i'd figured out how to make the login error alert more accessible.  but the end to end test seems to be failing consistently (or at least failed twice in the same way).  unexpected, since the test makes it past login, and fails on opening a resource template and subsequent steps.  will investigate further monday.

**Review by jgreben — COMMENTED** [review]
Looks good other than to get the build and end2end cypress test to pass and to rebase from master.

**Comment by jmartin-sul** [comment]
i don't now that this PR makes sense anymore, given that its primary purpose was a `findByRole` test (and [we've stopped using `ByRole`](https://github.com/LD4P/sinopia_editor/pull/2470) selectors for performance reasons).

closing...

**Pull request closed without merging**